Makefile

```diff
@@ -770,20 +770,29 @@ install.docker-full: install.docker install.docker-docs
 
 .PHONY: install.systemd
 ifneq (,$(findstring systemd,$(BUILDTAGS)))
-install.systemd:
+PODMAN_UNIT_FILES = contrib/systemd/auto-update/podman-auto-update.service \
+		    contrib/systemd/system/podman.service \
+		    contrib/systemd/system/podman-restart.service
+
+%.service: %.service.in
+	sed -e "s;@@PODMAN@@;$(BINDIR)/podman;g" $< >$@.tmp.$$ \
+		&& mv -f $@.tmp.$$ $@
```

Comment:
Dumb question: Why the separate `mv`?

[Non-blocking] Someone else might wonder this as well, if the reason is important, maybe it should be in the commit message or a comment?

Reply:
> because that is a recipe for disaster

I'm not trying to argue, just want to understand how this is likely to be bad:  Don't most build tool-chains simply output files directly in the repo (w/o a move)?  I could understand clashes if/when it outputting to a destination file on the system, but howso in this context? Is it in case `make` is running with `--jobs` or something like that?

Reply:
No, that was @vrothberg's first approach in #11788. It turned out to be nonworkable because 
1. it modifies git-controlled files, so the CI "clean working dir" test failed (and even if it hadn't, modifying git-controlled files is a super bad idea); and 
2.  it is nontransitive: if you run `make install.systemd` once, you can never run it again with a different path, because the Makefile rule has already replaced `/usr/bin/podman` with something else, so the search string is no longer in the source unit files.

Reply:
Regarging `> $@`, sorry, I was terse because I assumed common knowledge. Using `> $@` is something I will never do. It is evil. If you get disk full, the target is corrupt. If the generating tool crashes or fails, the target is corrupt. And in all those cases, you end up advertising that the target exists and was created successfully, and subsequent build steps will assume that it is safe to use it, but it isn't. The only possible correct way to do this is to write a tmp file, then, if everything works, you mv the temp file into place. (One must always do this everywhere, not just in Makefiles).

Reply:
Ohhh, okay, right that makes sense, since `make` only cares about file existence and timestamps.  Thanks.